platform/darwin/src/MGLLocationManager.h

```diff
@@ -0,0 +1,103 @@
+#import <Foundation/Foundation.h>
+#import <CoreLocation/CoreLocation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol MGLLocationManagerDelegate;
+
+/**
+ The `MGLLocationManager` protocol defines a set of methods that you
+ use to receive location-related events.
+ */
+@protocol MGLLocationManager <NSObject>
+
+@required
+
+/**
+ The delegate to receive updates.
+ */
+@property (nonatomic, weak) id<MGLLocationManagerDelegate> delegate;
+
+/**
+ Specifies a physical device orientation.
+ */
+@property (nonatomic) CLDeviceOrientation headingOrientation;
+
+/**
+ Returns the current localization authorization status.
```

Comment:
An implementer may skim this interfce and expect it to match CLLocationManager’s interface. But they’d overlook this subtle difference: `authorizationStatus` is a property (instance method) instead of a class method. Maybe we can clarify that.

Reply:
It may not be necessary because when you implement the protocol you get the warning.